test/e2e/suite/conformance/certificatesigningrequests/suite.go

```diff
@@ -92,15 +134,11 @@ func (s *Suite) complete(f *framework.Framework) {
 		Fail("CreateIssuerFunc must be set")
 	}
 
-	if s.DomainSuffix == "" {
-		s.DomainSuffix = f.Config.Addons.IngressController.Domain
-	}
-
-	if s.UnsupportedFeatures == nil {
-		s.UnsupportedFeatures = make(featureset.FeatureSet)
+	if s.HTTP01TestType == "Gateway" {
+		framework.RequireFeatureGate(utilfeature.DefaultFeatureGate, feature.ExperimentalGatewayAPISupport)
```

Comment:
We do not have `f *framework.Framework` here, but it was required by the `framework.RequireFeatureGate(` function (where it was not used).

Reply:
So, I updated the function, which added a lot of changes to this PR.